### ISSUE TITLE
Enhance jsx-uses-vars

### ISSUE THREAD

**Issue by miljan-aleksic** [vuejs/eslint-plugin-vue#193]
* **ESLint Version:** 4.7
* **eslint-plugin-vue Version:** 3.13
* **Node Version:** 8.5

**Please show your full configuration:**

```
module.exports = {
  root: true,
  parserOptions: {
    parser: 'babel-eslint',
    sourceType: 'module'
  },
  env: {
    browser: true,
    node: true
  },
  extends: [
    'eslint:recommended',
    'plugin:vue/recommended'
  ]
}
```

**What did you do?**

```vue
export default (h) => <MyComponent></MyComponent>
```

**What did you expect to happen?**

I expected to rise no errors.

**What actually happened? Please include the actual, raw output from ESLint.**

`error  'h' is defined but never used  no-unused-vars`

I know the $createElement args is auto-injected using latest versions but is not always the case and the arg could be required.

**Comment by michalsnik** [comment]
Hey, I get it @miljan-aleksic but it looks like it's only the case while using arrow function, because then you have to explicitly catch `createElement` argument. Fixing this would require us to mark first parameter in any arrow function as used, which I'm not sure would be a good thing.. you can instead use normal function expression and you won't need to catch the argument.

**Comment by miljan-aleksic** [comment]
Hi @michalsnik, thank you for getting back to me. I have tested it with normal function and the issue is the same.

```js
// throws an error
export default {
  functional: true,
  render (h) {
    return <MyComponent></MyComponent>
  }
}
```

But, funny, this other definition doesn't throw an error.

```js
// doesnt throw an error
export default {
  functional: true,
  render (h, { parent }) {
    console.log(parent) // eslint-disable-line
    return <MyComponent></MyComponent>
  }
}
```

Definitely, there is some bug about this and perhaps the right way to go would be to only mark the first argument as used when the name of the function is render and there is a return used.

**Comment by michalsnik** [comment]
Regarding first code example if you omit the ‘h’ it should work properly
without any error.

And the second doesn’t throw an error as it’s intended. You expect second
argument so you had to define first, but you don’t have to use to it.

**Comment by miljan-aleksic** [comment]
you are totally right @michalsnik :) Thank you.
